### PR TITLE
fix  "Timeout" on some pending download jobs

### PR DESCRIPTION
**Pull request by julien-louis**
https://github.com/fli-iam/shanoir-ng/issues/1988

manage interrupted downloads by keeping track of open tabs as 'sessions'. Each open browser tab (or 'session') have to give a regular life sign so the other ones know if a dl was interrupted.

### CODE REVIEW

File: shanoir-ng-front/src/app/async-tasks/async-tasks.component.ts [hunk 1]
```diff
@@ -85,6 +85,8 @@ export class AsyncTasksComponent extends EntityListComponent<Task> implements Af
     }
 
     select(lightTask: Task) {
+        this.notificationsService.nbNew = 0;
```
Comment (julien-louis) [inline shanoir-ng-front/src/app/async-tasks/async-tasks.component.ts:88]: removes the green bell when navigating into tasks

File: shanoir-ng-front/src/app/shared/mass-download/mass-download.service.ts [hunk 2]
```diff
@@ -526,6 +526,7 @@ export class MassDownloadService {
         task.progress = 0;
         task.status = 2;
         task.eventType = 'downloadDataset.event';
+        task.sessionId = this.sessionService.sessionId;
```
Comment (julien-louis) [inline shanoir-ng-front/src/app/shared/mass-download/mass-download.service.ts:529]: add the id of the current browser tab, so if it is closed we know the task is in error state

**Comment by pierrehenri-dauvergne** [comment]
J'ai testé en local et en qualif :
j'ai eu pas mal d'erreurs en local, genre ça :

> 2024-01-17T15:30:33.158Z ERROR 1 --- [nio-9904-exec-6] o.s.ng.download.WADODownloaderService    : A dicom file could not be downloaded from the pacs:
> datasets                  |
> datasets                  | java.util.zip.ZipException: duplicate entry: subject2_20100304_T1_VIBE3D_5__relaxo_1.4.9.12.34.1.8527.4213964376020221675332795001857180494616.dcm

Ca défile en boucle dans la console mais sur la page web, visuellement y'a pas d'erreurs sur firefox, alors que sur chrome j'ai le symbole /!\  sur le Job

J'ai testé sur la qualif car je suppose que c'est lié à mes datasets copiés et sur firefox ça a bien marché, sur chrome ça a marché jusqu'à la toute fin où j'ai eu cette erreur dans la console :

> InvalidStateError: An operation that depends on state cached in an interface object was made but the state had changed since it was read from disk

Et dans les logs j'ai :

> Caused by: org.hibernate.exception.DataException: could not execute statement [Data truncation: Data too long for column 'object_id' at row 1] [insert into events (creation_date,event_type,last_update,message,object_id,progress,report,status,study_id,user_id,id) values (?,?,?,?,?,?,?,?,?,?,?)]

**Comment by julien-louis** [comment]
@pierrehenri-dauvergne I think all those bugs are addressed in https://github.com/fli-iam/shanoir-ng/pull/2015

maybe this one should be tested again when the #2015 is merged.

**Comment by pierrehenri-dauvergne** [comment]
Tested in qualif:

Multiple tests were OK but the tasks last less than 5mn, since this PR is to make sure the job doesn't end in timeout, I selected more datasets (longer download). The first time I had the issue below, the second time (with the next 200 datasets) everything went well. Maybe the issue is caused by my computer (lack of disk space or something?)

How I tested:
- Filtered by study (UCAN)
- Picked the first 200 datasets (ids found in log below) and got this error:
"error while downloading : End of data reached (data length = 2349083367, asked index = -1948022935). Corrupted zip ?"


Logs says: 
2024-03-12T16:47:14.071+01:00  INFO 1 --- [org.springframework.amqp.rabbit.RabbitListenerEndpointContainer#4-1] o.s.n.c.amqp.RabbitMQUserService         : Receiving event: {"id":8999669871841397441,"eventType":"downloadDataset.event","objectId":"498846,498847,498848,498849,498850,498851,498852,498853,498854,498855,498856,498857,498858,498859,498860,498861,498862,500457,500458,500459,500460,500461,500462,500463,500464,500465,500466,500467,500468,500469,500470,502840,502841,502842,502843,502844,502845,502846,502847,502848,502849,502850,502851,502852,502853,502854,502855,502856,502857,502858,502859,502860,502861,502862,502863,502864,502865,502866,502867,502868,502869,502870,502871,502872,502873,502874,502875,502876,502877,502878,502879,502880,502881,502882,502883,502884,502885,502886,502887,502888,502889,502890,502891,502892,502893,502894,502895,503892,503893,503894,503895,503896,503897,503898,503899,503900,503901,503902,503903,503904,503905,503906,503907,503908,503909,503910,503911,503912,503913,503914,503915,503916,503917,503918,503919,503920,503921,503922,503923,503924,503925,503926,504509,504510,504511,504512,504513,504514,504515,504516,504517,504518,504519,504520,504521,504522,504523,504524,504525,504526,504527,504528,504529,504530,504531,504532,504533,504534,504535,504536,504537,504538,504539,504540,504541,504542,504543,504544,504545,504546,504547,504548,504549,504550,504551,504552,504553,504554,504555,504556,504557,504558,504559,504560,504561,504562,504563,504564,504565,504566,504567,504568,504569,504570,504571,504572,504573,504574,504575,504576,504577,504578,504579,504580,504581,504582,504583,504584,504585,504586","userId":1120,"message":"498846,498847,498848,498849,498850,498851,498852,498853,498854,498855,498856,498857,498858,498859,498860,498861,498862,500457,500458,500459,500460,500461,500462,500463,500464,500465,500466,500467,500468,500469,500470,502840,502841,502842,502843,502844,502845,502846,502847,502848,502849,502850,502851,502852,502853,502854,502855,502856,502857,502858,502859,502860,502861,502862,502863,502864,502865,502866,502867,502868,502869,502870,502871,502872,502873,502874,502875,502876,502877,502878,502879,502880,502881,502882,502883,502884,502885,502886,502887,502888,502889,502890,502891,502892,502893,502894,502895,503892,503893,503894,503895,503896,503897,503898,503899,503900,503901,503902,503903,503904,503905,503906,503907,503908,503909,503910,503911,503912,503913,503914,503915,503916,503917,503918,503919,503920,503921,503922,503923,503924,503925,503926,504509,504510,504511,504512,504513,504514,504515,504516,504517,504518,504519,504520,504521,504522,504523,504524,504525,504526,504527,504528,504529,504530,504531,504532,504533,504534,504535,504536,504537,504538,504539,504540,504541,504542,504543,504544,504545,504546,504547,504548,504549,504550,504551,504552,504553,504554,504555,504556,504557,504558,504559,504560,504561,504562,504563,504564,504565,504566,504567,504568,504569,504570,504571,504572,504573,504574,504575,504576,504577,504578,504579,504580,504581,504582,504583,504584,504585,504586.dcm","report":null,"status":1,"progress":null,"studyId":null,"timestamp":null}
2024-03-12T16:47:14.078+01:00 ERROR 1 --- [org.springframework.amqp.rabbit.RabbitListenerEndpointContainer#4-1] o.h.engine.jdbc.spi.SqlExceptionHelper   : Data truncation: Data too long for column 'object_id' at row 1
2024-03-12T16:47:14.079+01:00 ERROR 1 --- [org.springframework.amqp.rabbit.RabbitListenerEndpointContainer#4-1] o.s.n.c.amqp.RabbitMQUserService         : Something went wrong deserializing the event.

**Review by pierrehenri-dauvergne — APPROVED** [review]

**Pull request merged**